File: Software/workspace/SimTest/src/Commands/Autonomous.cpp
/*
 * Autonomous.cpp
 *
 *  Created on: Mar 3, 2016
 *      Author: alpiner
 */

#include <Commands/Autonomous.h>
#include <Commands/DriveStraight.h>
#include <Commands/ExecLoader.h>
#include <Commands/CloseGate.h>
#include <Commands/OpenGate.h>
#include <Commands/StepShooterAngle.h>
#include <Commands/ShootBall.h>
#include <Commands/Turn.h>
#include <Commands/FullStop.h>
#include "Robot.h"

Autonomous::Autonomous() : CommandGroup("Autonomous") {
	AddSequential(new CloseGate()); // hold the ball
	AddSequential(new OpenGate()); // pinch the ball
	AddSequential(new DriveStraight(7,0)); // go forward
	AddSequential(new Turn(-50)); // turn
	AddSequential(new StepShooterAngle(36)); // set angle
	AddSequential(new ShootBall()); // shoot
	AddSequential(new FullStop()); // end autonomous
}

void Autonomous::Interrupted() {
	std::cout << "Autonomous::Interruped"<<std::endl;
	_End();
}

void  Autonomous::Cancel(){
	std::cout << "Autonomous::Cancel"<<std::endl;
	_End();
	CommandGroup::Cancel();
	Robot::drivetrain->EndTravel();
}

File: Software/workspace/JSim2018Bot/src/org/usfirst/frc/team159/robot/commands/Autonomous.java
package org.usfirst.frc.team159.robot.commands;

import org.usfirst.frc.team159.robot.Robot;
import edu.wpi.first.wpilibj.command.CommandGroup;

/**
 *
 */
public class Autonomous extends CommandGroup {

    public Autonomous() {
		requires(Robot.driveTrain);
    	//addSequential(new DriveStraight(5.0));
    //	addSequential(new DrivePath());

    }
}
